ql/termstructures/volatility/equityfx/andreasenhugevolatilityadapter.cpp

```diff
@@ -0,0 +1,65 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2017 Klaus Spanderen
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+#include <ql/math/functional.hpp>
+#include <ql/pricingengines/blackformula.hpp>
+#include <ql/termstructures/yieldtermstructure.hpp>
+#include <ql/termstructures/volatility/equityfx/andreasenhugevolatilityinterpl.hpp>
+#include <ql/termstructures/volatility/equityfx/andreasenhugevolatilityadapter.hpp>
+
+namespace QuantLib {
+
+    AndreasenHugeVolatilityAdapter::AndreasenHugeVolatilityAdapter(
+        const boost::shared_ptr<AndreasenHugeVolatilityInterpl>& volInterpl,
+        Real eps)
+    : BlackVarianceTermStructure(
```

Comment:
Instead of copying the arguments below, you might also use the default constructor and override `referenceDate`, `calendar` and the other relevant methods.  This would remain correct in the case the reference date of the risk-free curve moves; the current implementation doesn't update the volatility reference date accordingly.  (I'm not sure if the same goes for the local-vol adapter, since you don't seem to copy anything there...)

Reply:
Added the missing bit, please see the corresponding test case.